src/index.ts

```diff
@@ -266,14 +266,18 @@ export async function signAddon({
     }
   }
 
-  signedFile = await poll(
-    async () => {
-      const file = await client.getSignedFile(addonId, addonVersion);
-      if (!file) throw new Error('file not signed yet');
-      return file;
-    },
-    pollInterval,
-    pollRetry
-  );
-  return client.downloadFile(signedFile.download_url, output);
+  if (!output && channel === 'unlisted') {
+    throw new Error('unlisted addons must have an output');
+  } else if (output) {
```

Comment:
Same problem, harder for automation because the returned value might be undefined. I prefer throwing an error instead.

Reply:
nit: `else` is not needed here.